Aws/S3/Core.hs

```diff
@@ -49,6 +53,16 @@ data RequestStyle
     | VHostStyle
     deriving (Show)
 
+data S3SignPayloadMode
+    = AlwaysUnsigned -- ^ Always use the "UNSIGNED-PAYLOAD" option.
+    | SignWithEffort -- ^ Sign the payload when 'HTTP.RequestBody' is a on-memory one ('HTTP.RequestBodyLBS' or 'HTTP.RequestBodyBS'). Otherwise use the "UNSINGED-PAYLOAD" option.
```

Comment:
Perhaps add a third option which errors out instead of making an unsigned payload?

Reply:
I have added it.